src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2752,6 +2752,32 @@ private unsafe void SendManifest(byte[]? rawManifest)
 #endif // FEATURE_MANAGED_ETW
         }
 
+        // Helper to deal with the fact that the type we are reflecting over might be loaded in the ReflectionOnly context.
+        // When that is the case, we have to build the custom assemblies on a member by hand.
+        internal static bool IsCustomAttributeDefinedHelper(
+            MemberInfo member,
+            Type attributeType,
+            EventManifestOptions flags = EventManifestOptions.None)
+        {
+            // AllowEventSourceOverride is an option that allows either Microsoft.Diagnostics.Tracing or
+            // System.Diagnostics.Tracing EventSource to be considered valid. This should not matter anywhere but in Microsoft.Diagnostics.Tracing (nuget package).
```

Comment:
Then should it be ifdef'd out of corelib?

Reply:
I copied it from `GetCustomAttributeHelper` just below; not sure about what it means in practice?